# <center>Домашнє завдання № 1 (демо).<br> Аналіз даних доходів населення UCI Adult</center>

**В завданні пропонується за допомогою Pandas дати відповідь на кілька запитань по даним репозиторія UCI [Adult](https://archive.ics.uci.edu/ml/datasets/Adult).**

Унікальні значення ознак:
- age: continuous.
- workclass: Private, Self-emp-not-inc, Self-emp-inc, Federal-gov, Local-gov, State-gov, Without-pay, Never-worked.
- fnlwgt: continuous.
- education: Bachelors, Some-college, 11th, HS-grad, Prof-school, Assoc-acdm, Assoc-voc, 9th, 7th-8th, 12th, Masters, 1st-4th, 10th, Doctorate, 5th-6th, Preschool.
- education-num: continuous.
- marital-status: Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse.
- occupation: Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces.
- relationship: Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried.
- race: White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black.
- sex: Female, Male.
- capital-gain: continuous.
- capital-loss: continuous.
- hours-per-week: continuous.
- native-country: United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala, Nicaragua, Scotland, Thailand, Yugoslavia, El-Salvador, Trinadad&Tobago, Peru, Hong, Holand-Netherlands.   
- salary: >50K,<=50K

In [23]:
import pandas as pd
import numpy as np

In [2]:
data = pd.read_csv('adult.data.csv')
data.head()

,age,workclass,fnlwgt,education,education-num,marital-status,occupation,relationship,race,sex,capital-gain,capital-loss,hours-per-week,native-country,salary
0,39,State-gov,77516,Bachelors,13,Never-married,Adm-clerical,Not-in-family,White,Male,2174,0,40,United-States,<=50K
1,50,Self-emp-not-inc,83311,Bachelors,13,Married-civ-spouse,Exec-managerial,Husband,White,Male,0,0,13,United-States,<=50K
2,38,Private,215646,HS-grad,9,Divorced,Handlers-cleaners,Not-in-family,White,Male,0,0,40,United-States,<=50K
3,53,Private,234721,11th,7,Married-civ-spouse,Handlers-cleaners,Husband,Black,Male,0,0,40,United-States,<=50K
4,28,Private,338409,Bachelors,13,Married-civ-spouse,Prof-specialty,Wife,Black,Female,0,0,40,Cuba,<=50K


**1. Скільки чоловіків и жінок (ознака *sex*) представлено в цьому наборі даних?**

In [3]:
data['sex'].value_counts(dropna=False)

sex
Male      21790
Female    10771
Name: count, dtype: int64

Відповідь: 21790 чоловіків, 10771 жінок

**2. Який середній вік (ознака *age*) жінок?**

In [4]:
age_mean = data[data['sex'] == 'Female']['age'].mean()
print(age_mean.round(2))

36.86


**3. Яка доля громадян Німеччини (ознака *native-country*)?**

In [34]:
(data['native-country'] == 'Germany').mean()

np.float64(0.004207487485028101)

Відповідь: ≈0.004

**4-5. Які середні значення та средньоквадратичні відхилення віку тих, хто отримує більше 50K на рік (ознака *salary*) і тих, хто отримує менше 50K на рік?**

In [6]:
age_avg_1 = data[data['salary'] == '>50K']['age'].mean()
age_std_1 = data[data['salary'] == '>50K']['age'].std()
age_avg_2 = data[data['salary'] == '<=50K']['age'].mean()
age_std_2 = data[data['salary'] == '<=50K']['age'].std()
print(age_avg_1)
print(age_std_1)
print(age_avg_2)
print(age_std_2)

44.24984058155847
10.519027719851826
36.78373786407767
14.02008849082488


In [ ]:
Відповідь: вік тих, хто заробляє більше 50К: 44.25 ± 10.52
           вік тих, хто заробляє менше 50К: 36.78 ± 14.02

**6. Чи правда, що люди, які отримують більше 50k, мають як мінімум вищу освіту? (ознака *education – Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters* или *Doctorate*)**

In [7]:
data_1 = data[['salary', 'education']][data['salary'] == '>50K']
data_1['education'].isin(['Bachelors', 'Prof-school', 'Assoc-acdm', 'Assoc-voc', 'Masters', 'Doctorate']).mean()

np.float64(0.5783701058538452)

In [ ]:
Відповідь: оскільки лише приблизно 58 відсотків людей, що заробляють більше 50К, мають вищу освіту, то це твердження є хибним.

**7. Порахуйте статистику віку для кожної раси (ознака *race*) і кожної статі. Використайте *groupby* и *describe*. Знайдіть таким сином максимальний вік чоловіків раси *Amer-Indian-Eskimo*.**

In [29]:
data.groupby(['race', 'sex'])['age'].describe()

count       mean        std   min   25%   50%  \
race               sex                                                       
Amer-Indian-Eskimo Female    119.0  37.117647  13.114991  17.0  27.0  36.0   
                   Male      192.0  37.208333  12.049563  17.0  28.0  35.0   
Asian-Pac-Islander Female    346.0  35.089595  12.300845  17.0  25.0  33.0   
                   Male      693.0  39.073593  12.883944  18.0  29.0  37.0   
Black              Female   1555.0  37.854019  12.637197  17.0  28.0  37.0   
                   Male     1569.0  37.682600  12.882612  17.0  27.0  36.0   
Other              Female    109.0  31.678899  11.631599  17.0  23.0  29.0   
                   Male      162.0  34.654321  11.355531  17.0  26.0  32.0   
White              Female   8642.0  36.811618  14.329093  17.0  25.0  35.0   
                   Male    19174.0  39.652498  13.436029  17.0  29.0  38.0   

                             75%   max  
race               sex                  
Amer-Indian-Eskimo Female  46.00  80.0  
                   Male    45.00  82.0  
Asian-Pac-Islander Female  43.75  75.0  
                   Male    46.00  90.0  
Black              Female  46.00  90.0  
                   Male    46.00  90.0  
Other              Female  39.00  74.0  
                   Male    42.00  77.0  
White              Female  46.00  90.0  
                   Male    49.00  90.0

In [ ]:
Відповідь: 82 роки 

**8. Серед кого більша доля багатіших (>50K): серед одружених чи не одружених чоловіків (ознака *marital-status*)? За одружених вважаємо тих, у кого *marital-status* починається із *Married* (Married-civ-spouse, Married-spouse-absent или Married-AF-spouse), всі інші - не одружені.**

In [8]:
data_marriage = data[['salary', 'marital-status']][(data['sex'] == 'Male')]
data_married = data_marriage[data_marriage['marital-status'].isin(['Married-civ-spouse', 'Married-spouse-absent', 'Married-AF-spouse'])].value_counts('salary', normalize=True)
data_else = data_marriage[~ data_marriage['marital-status'].isin(['Married-civ-spouse', 'Married-spouse-absent', 'Married-AF-spouse'])].value_counts('salary', normalize=True)
print(data_married)
print(data_else)

salary
<=50K    0.559486
>50K     0.440514
Name: proportion, dtype: float64
salary
<=50K    0.915505
>50K     0.084495
Name: proportion, dtype: float64


In [ ]:
Відповідь: доля багатіших людей серед одружених чоловіків

**9. Яку максимальну кількість годин в тиждень працює людина (ознака *hours-per-week*)? Скільки людей працюють таку кількість годин і який відсоток серед них багатих?**

In [31]:
max_hours = data['hours-per-week'].max()
print('Кількість годин:', max_hours)
people_data = data[data['hours-per-week'] == max_hours]['salary']
print('Кількість людей: ', len(people_data))
data[data['hours-per-week'] == max_hours]['salary'].value_counts(normalize=True)

Кількість годин: 99
Кількість людей:  85


salary
<=50K    0.705882
>50K     0.294118
Name: proportion, dtype: float64

In [ ]:
Відповідь: так працюють 85 людей, приблизно 29 відсотків з них багаті.

**10. Порахуйте середню тривалість роботи на тиждень (*hours-per-week*) серед багатих і бідних (*salary*) для кожної країни (*native-country*).**

In [27]:
data.groupby(['native-country', 'salary'])[['hours-per-week']].agg([np.mean])

hours-per-week
                                mean
native-country salary               
?              <=50K       40.164760
               >50K        45.547945
Cambodia       <=50K       41.416667
               >50K        40.000000
Canada         <=50K       37.914634
...                              ...
United-States  >50K        45.505369
Vietnam        <=50K       37.193548
               >50K        39.200000
Yugoslavia     <=50K       41.600000
               >50K        49.500000

[82 rows x 1 columns]